pyvista/core/_validation/check.py

```diff
@@ -118,27 +158,28 @@ def check_real(arr, /, *, name="Array"):
     >>> _validation.check_real([1, 2, 3])
 
     """
-    arr = arr if isinstance(arr, np.ndarray) else _cast_to_numpy(arr)
-
-    # Return early for common cases
-    if arr.dtype.type in [np.int32, np.int64, np.float32, np.float64]:
-        return
-
-    # Do not use np.isreal as it will fail in some cases (e.g. scalars).
-    # Check dtype directly instead
+    dtype = _ArrayLikeWrapper(array).dtype
     try:
-        check_subdtype(arr, (np.floating, np.integer), name=name)
+        check_subdtype(dtype, (np.floating, np.integer), name=name)
     except TypeError as e:
         raise TypeError(f"{name} must have real numbers.") from e
 
 
-def check_sorted(arr, /, *, ascending=True, strict=False, axis=-1, name="Array"):
+def check_sorted(
+    array: _ArrayLikeOrScalar[_NumberType],
+    /,
```

Comment:
We could drop the `\` from all functions, there's no technical need to have it like this (I think its use stems from compatibility with C). Just having `*` for kwargs only is OK I think.

The [example from scientific python](https://learn.scientific-python.org/development/principles/design/#complexity-is-always-conserved) has `kwargs` only as a recommendation.

Reply:
I think the current use of `/` and `*` is fine.